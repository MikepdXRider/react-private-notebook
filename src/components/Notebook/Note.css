.subtitle {
  font-size: small;
  color: var(--light-gray);
}
