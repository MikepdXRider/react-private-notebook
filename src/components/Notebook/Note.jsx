import { Link } from 'react-router-dom';
import styles from './Note.css';

export default function Note({ note }) {
  return (
    <article>
      <h2>{note.title}</h2>
      <p className={styles.subtitle}>
        Created {note.createdAt}
        {note.updatedAt && <span> â€” Last updated {note.updatedAt}</span>}
      </p>
      <Link to={`/notes/${note.id}/edit`}>
        <button>Edit</button>
      </Link>
      <hr />
      <p>{note.content}</p>
    </article>
  );
}
