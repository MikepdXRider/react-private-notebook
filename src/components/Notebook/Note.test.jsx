import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Note from './Note';

it('should display the provided note', () => {
  const { container } = render(
    <MemoryRouter>
      <Note
        note={{
          id: 'id',
          title: 'title',
          content: 'content',
          createdAt: 'createdAt',
          updatedAt: 'updatedAt',
        }}
      />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
