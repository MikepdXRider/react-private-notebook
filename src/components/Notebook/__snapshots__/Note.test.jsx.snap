// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the provided note 1`] = `
<div>
  <article>
    <h2>
      title
    </h2>
    <p
      class="subtitle"
    >
      Created 
      createdAt
      <span>
         â€” Last updated 
        updatedAt
      </span>
    </p>
    <a
      href="/notes/id/edit"
    >
      <button>
        Edit
      </button>
    </a>
    <hr />
    <p>
      content
    </p>
  </article>
</div>
`;
